ui/pages/remote-mode/hooks/useRemoteMode.ts

```diff
@@ -0,0 +1,174 @@
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import { TransactionType } from '@metamask/transaction-controller';
+import { type Hex, hexToNumber } from '@metamask/utils';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  createDelegation,
+  getDeleGatorEnvironment,
+} from '../../../../shared/lib/delegation';
+import {
+  encodeDisableDelegation,
+  getDelegationHashOffchain,
+} from '../../../../shared/lib/delegation/delegation';
+import { getSelectedNetworkClientId } from '../../../../shared/modules/selectors/networks';
+import { getSelectedNetwork } from '../../../selectors';
+import { getRemoteModeConfig } from '../../../selectors/remote-mode';
+import { addTransaction } from '../../../store/actions';
+import {
+  awaitDeleteDelegationEntry,
+  listDelegationEntries,
+  signDelegation,
+  storeDelegationEntry,
+} from '../../../store/controller-actions/delegation-controller';
+import { useEIP7702Account } from '../../confirmations/hooks/useEIP7702Account';
+import { useEIP7702Networks } from '../../confirmations/hooks/useEIP7702Networks';
+import { REMOTE_MODES } from '../remote.types';
+import { useConfirmationNavigation } from '../../confirmations/hooks/useConfirmationNavigation';
+
+export const useRemoteMode = ({ account }: { account: Hex }) => {
```

Comment:
This could also be a future refactor, but I feel this hook might be handling a few too many concerns and could benefit from being split into more focussed hooks...would make for easier maintenance, testability, error handling (per the previous comment), etc. wdyt?

Reply:
How would you suggest splitting up?